Update migration for SignalR (#12305)
aspnet/AspNetCore.Docs: aspnetcore/migration/22-to-30.md

## Update SignalR code

> System.Text.Json is now the default Hub Protocol used by both the client and server.

Reviewer: Might be helpful to link to an announcement/docs here.

> System.Text.Json is now the default Hub Protocol used by both the client and server.

Reviewer 2: ```suggestion
`System.Text.Json` is now the default Hub Protocol used by both the client and server.
```

> Use `AddJsonProtocol` to set serializer options.

Reviewer 2: ```suggestion
In `Startup.ConfigureServices`, call `AddJsonProtocol` to set serializer options.
```

> Server:

Reviewer 3: Can we use headers for "Server:" and "Client:"?

> Server:

Reviewer 2: ```suggestion
**Server:**

```

```csharp
services.AddSignalR(...)
        .AddJsonProtocol(options =>
        {

>             options.WriteIndented = false;

Reviewer 3: We should link to the AddJsonProtocol/JsonHubProtocolOptions API docs once they're updated.

        })
```

> Client:

Reviewer 2: ```suggestion
**Client:**

```

```csharp
new HubConnectionBuilder()
    .WithUrl("/chatHub")
    .AddJsonProtocol(options =>
    {
        options.WriteIndented = false;
    })
    .Build();
```

> If you're using features of `Newtonsoft.Json` that aren't supported in `System.Text.Json`, you can switch back to `Newtonsoft.Json`:

Reviewer 3: The section to switch back to Newtonsoft.Json probably also deserves its own subheading.

> If you were previously using `AddJsonProtocol` with Newtonsoft.Json options or want to continue using Newtonsoft.Json instead of the new default the following instructions will show you how:

Reviewer: This almost feels like it warrants a small heading to the effect of "Using Json.NET in SignalR" or "Continuing to use Json.NET in SignalR"

> If you were previously using `AddJsonProtocol` with Newtonsoft.Json options or want to continue using Newtonsoft.Json instead of the new default the following instructions will show you how:

Author: Can't code suggest your code suggest, but "are'nt" -> "aren't"

> If you were previously using `AddJsonProtocol` with Newtonsoft.Json options or want to continue using Newtonsoft.Json instead of the new default the following instructions will show you how:

Reviewer 4: a'int

> If you were previously using `AddJsonProtocol` with Newtonsoft.Json options or want to continue using Newtonsoft.Json instead of the new default the following instructions will show you how:

Reviewer 4: amn't

> 1. Install the `Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson` NuGet package.

Reviewer 2: ```suggestion
1. Install the [Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson](https://www.nuget.org/packages/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson) NuGet package.
```

> 1. Install the Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson NuGet package.

Reviewer: ```suggestion
1. Install the `Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson` NuGet package.
```

> 2. On the client add `.AddNewtonsoftJsonProtocol()` to the `HubConnectionBuilder`:

Reviewer 2: ```suggestion
1. On the client, chain an `AddNewtonsoftJsonProtocol` method call to the `HubConnectionBuilder` instance:

```

```csharp
new HubConnectionBuilder()
    .WithUrl("/chatHub")
    .AddNewtonsoftJsonProtocol(...)
    .Build();
```

> 3. On the server add `.AddNewtonsoftJsonProtocol()` to the `AddSignalR()` call:

Reviewer 2: ```suggestion

3. On the server, chain an `AddNewtonsoftJsonProtocol` method call to the `AddSignalR` method call in `Startup.ConfigureServices`:

```

## Update SignalR code

`System.Text.Json` is now the default Hub Protocol used by both the client and server.

In `Startup.ConfigureServices`, call `AddJsonProtocol` to set serializer options.

**Server:**

```csharp
services.AddSignalR(...)
        .AddJsonProtocol(options =>
        {
            options.WriteIndented = false;
        })
```

**Client:**

```csharp
new HubConnectionBuilder()
    .WithUrl("/chatHub")
    .AddJsonProtocol(options =>
    {
        options.WriteIndented = false;
    })
    .Build();
```

### Switch to Newtonsoft.Json
If you're using features of `Newtonsoft.Json` that aren't supported in `System.Text.Json`, you can switch back to `Newtonsoft.Json`:

1. Install the [Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson](https://www.nuget.org/packages/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson) NuGet package.
1. On the client, chain an `AddNewtonsoftJsonProtocol` method call to the `HubConnectionBuilder` instance:

```csharp
new HubConnectionBuilder()
    .WithUrl("/chatHub")
    .AddNewtonsoftJsonProtocol(...)
    .Build();
```

> 3. On the server, chain an `AddNewtonsoftJsonProtocol` method call to the `AddSignalR` method call in `Startup.ConfigureServices`:

Reviewer 2: ```suggestion
1. On the server, chain an `AddNewtonsoftJsonProtocol` method call to the `AddSignalR` method call in `Startup.ConfigureServices`:
```

Author: This wasn't working before because of the code snippet. Maybe it works now that there is a newline?

Author: Nope, reverted the change
